databricks/azure_auth.go

```diff
@@ -213,53 +212,11 @@ func (a *AzureAuth) initWorkspaceAndGetClient(config *service.DBApiClientConfig)
 	//}
 	dbClient.SetConfig(&newOption)
 
-	// So when the workspace is initially created sometimes it fails to perform api calls so this is a simple test
-	// to verify that the workspace has been created successfully. The retry is intentional as sometimes after workspace
-	// creation the API's will not work correctly. This may also
-	err = validateWorkspaceApis(10, 30, func(attempt int) error {
-		_, err = dbClient.Clusters().ListNodeTypes()
-		return err
-	})
+	// Spin for a while while the workspace comes up and starts behaving.
+	_, err = dbClient.Clusters().ListNodeTypes()
```

Comment:
With the changes made it may be worth removing this arbitrary check at the provider configuration. This was pretty much a hack to verify if the workspace is setup properly. With the improvements made in this PR to use the hashicorp httpretry it may be worth removing lines 215 - 219. We may want to just let the resource object fail with an invalid token rather than the provider configure failure. 

I am not sure if was the intent of the provider configure to verify that the client object will throw a 403. Any thoughts on this @stuartleeks.

Reply:
We added this check back in as it got us past a few remaining errors :-)